metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2180,6 +2180,16 @@ public void addDynamicPartitions(long txnId, String dbName, String tableName,
     client.add_dynamic_partitions(adp);
   }
 
+  @Override
+  public int insertCommit(Table table, boolean overwrite) {
+    try {
+      getHook(table).commitInsert(table, overwrite);
```

Comment:
If storage handler for the table does not implement the hook, we will get an NPE here. We need a _hook != null_ check, as for the _createTable_ method.